### PR TITLE
Catch up to changing Capistrano 3.5.0 with specs

### PR DESCRIPTION
**Pull request by dshmelev**
Capistrano 3.5.0 has many changes.

https://github.com/capistrano/capistrano/blob/master/CHANGELOG.md#350
In order to notify Slack using Slackistrano, I think that it is necessary to catch up with some changes.

Thanks you @machupicchubeta

### CODE REVIEW

**Comment by machupicchubeta** [comment]
👍

**Comment by zstarer** [comment]
this worked for me 👍 !

**Comment by nimashariatian** [comment]
this is great! can you please review and merge?

**Comment by phallstrom** [comment]
Soon. Need to see how this will affect folks using older versions of Capistrano. Don't want to [totally] yank the rug out from them. 

Mostly just need to find some time to do it 😀

**Comment by nimashariatian** [comment]
great, thank you!

File: spec/dry_run_spec.rb [hunk 1]
```diff
@@ -1,10 +1,6 @@
 require 'spec_helper'
 
 describe Slackistrano do
-  before(:each) do
-    Rake::Task['load:defaults'].execute
```
Comment (phallstrom) [inline spec/dry_run_spec.rb]: How are you running the specs? I don't get any errors when running `rspec spec/dry_run_spec.rb`.

File: spec/dry_run_spec.rb [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
+      set "slack_run_#{stage}".to_sym, -> { true }
+      allow(::Capistrano::Configuration).to receive(:dry_run?).and_return true
+      expect(Slackistrano::Capistrano).not_to receive(:post_to_slack)
+      Rake::Task["slack:deploy:#{stage}"].execute
+    end
+  end
+
   %w(updating reverting updated reverted failed).each do |stage|
-    it "does not post to slack on slack:deploy:#{stage}" do
+    it "post to slack on slack:deploy:#{stage} without dry-run" do
```
Comment (phallstrom) [inline spec/dry_run_spec.rb]: I like this, but think I'll leave it out as the rest of the tests exercise the non-dry-run paths.

File: spec/dry_run_spec.rb [hunk 3]
```diff
@@ -1,12 +1,11 @@
 require 'spec_helper'
 
 describe Slackistrano do
-  %w[updating reverting updated reverted failed].each do |stage|
+  %w(updating reverting updated reverted failed).each do |stage|
```
Comment (phallstrom) [inline spec/dry_run_spec.rb]: Heh. I can't stand this style of arrays unfortunately.  I probably should add rubocop to the project in general, but going to leave this one out for now.  Grabbed the other one though. Thx.

Comment (machupicchubeta) [inline spec/dry_run_spec.rb]: Thank you for telling us your style. 
I think that makes sense.

Would you mind if I revert this line change? or Would you revert this by yourself?

File: spec/dry_run_spec.rb (hunk 1 above)
Comment (machupicchubeta) [inline spec/dry_run_spec.rb]: Thank you for pointed them out!

Hmm, I ran the spec following commands, and it was errors.

But .... now, I look it in more specifically, think this commit is unnecessary. :scream_cat:

```
$ bundle exec rspec spec/dry_run_spec.rb 
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 42008

Slackistrano
** Execute load:defaults
** Execute slack:deploy:updating
WARN [slackistrano] Error notifying Slack!
WARN [slackistrano]   Error: #<SocketError: Failed to open TCP connection to .slack.com:443 (getaddrinfo: nodename nor servname provided, or not known)>
  does not post to slack on slack:deploy:updating (FAILED - 1)

Failures:

  1) Slackistrano does not post to slack on slack:deploy:updating
     Failure/Error: if response.code !~ /^2/
     
     NoMethodError:
       undefined method `code' for nil:NilClass
     # ./lib/slackistrano/capistrano.rb:92:in `block in post'
     # ./lib/slackistrano/capistrano.rb:67:in `each'
     # ./lib/slackistrano/capistrano.rb:67:in `post'
     # ./lib/slackistrano/capistrano.rb:29:in `block in run'
     # ./.bundle/ruby/2.3.0/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:29:in `instance_exec'
     # ./.bundle/ruby/2.3.0/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:29:in `run'
     # ./.bundle/ruby/2.3.0/gems/capistrano-3.5.0/lib/capistrano/dsl.rb:58:in `run_locally'
     # ./lib/slackistrano/capistrano.rb:29:in `run'
     # ./lib/slackistrano/tasks/slack.rake:6:in `block (3 levels) in <top (required)>'
     # ./.bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/task.rb:248:in `block in execute'
     # ./.bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/task.rb:243:in `each'
     # ./.bundle/ruby/2.3.0/gems/rake-11.1.2/lib/rake/task.rb:243:in `execute'
     # ./spec/dry_run_spec.rb:12:in `block (3 levels) in <top (required)>'

Finished in 0.26051 seconds (files took 3.17 seconds to load)
1 example, 1 failure

Failed examples:

rspec './spec/dry_run_spec.rb[1:1]' # Slackistrano does not post to slack on slack:deploy:updating

Randomized with seed 42008
```

**Comment by machupicchubeta** [comment]
I think that the essence of this Pull Request has been imported into Slackistrano Version 3.0.0.
Thank you @phallstrom

**Comment by phallstrom** [comment]
@machupicchubeta I believe so too.  Appreciate the 3.5.0 dry run stuff. I knew that was coming, but wasn't sure when it would hit.  Thanks much!

**Pull request closed without merging**